app/collections/settings.jsx

```diff
@@ -89,35 +88,18 @@ export default class CollectionSettings extends React.Component {
       });
   }
 
-  handleDescriptionInputChange() {
-    const property = 'description';
-    const setting = this.state.setting;
-    setting[property] = true;
-    this.setState({ error: null, setting });
-
-    const descriptionText = this.description.value;
+  handleDescriptionInputChange(event) {
```

Comment:
This whole method could be a bit more concise and readable. 
This can be reduced to:
```
handleDescriptionInputChange(event) {
  const description = event.target.value;
  this.setState({ description })

  if (description.length <= 300) {
     this.props.collection.update({ description });
  }
}
```

Using the name attribute from an input is useful, but not necessary here since we're only concerned with updating the description.